interface/modules/custom_modules/oe-module-dorn/public/lab_setup.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/**
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ *
+ * @author    Brad Sharp <brad.sharp@claimrev.com>
+ * @copyright Copyright (c) 2022 Brad Sharp <brad.sharp@claimrev.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once "../../../../globals.php";
+
+use OpenEMR\Common\Acl\AclMain;
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Twig\TwigContainer;
+use OpenEMR\Core\Header;
+use OpenEMR\Modules\Dorn\ConnectorApi;
+
+//this is needed along with setupHeader() to get the pop up to appear
+
+$tab = "lab setup";
+if (!AclMain::aclCheckCore('admin', 'users')) {
+    echo (new TwigContainer(null, $GLOBALS['kernel']))->getTwig()->render('core/unauthorized.html.twig', ['pageTitle' => xl("Edit/Add Procedure Provider")]);
+    exit;
+}
+
+if (!empty($_POST)) {
+    if (isset($_POST['SubmitButton'])) {
```

Comment:
verifyCsrfToken check?

Reply:
This is a form called from menu and not ajax.